js/modules/k6/experimental/fs/module_test.go

```diff
@@ -212,6 +212,133 @@ func TestFile(t *testing.T) {
 
 		assert.NoError(t, err)
 	})
+
+	t.Run("read called with invalid argument should fail", func(t *testing.T) {
+		t.Parallel()
+
+		runtime, err := newConfiguredRuntime(t)
+		require.NoError(t, err)
+
+		testFilePath := fsext.FilePathSeparator + "bonjour.txt"
+		fs := newTestFs(t, func(fs afero.Fs) error {
+			return afero.WriteFile(fs, testFilePath, []byte("Bonjour, le monde"), 0o644)
+		})
+		runtime.VU.InitEnvField.FileSystems["file"] = fs
+
+		_, err = runtime.RunOnEventLoop(fmt.Sprintf(`
+			fs.open(%q).then(file => {
+				// No buffer argument provided should fail with TypeError.
+				file.read().then(
+					res => { throw 'unexpected promise resolution with result: ' + res },
+					err => { if (err.name !== 'TypeError') { throw 'unexpected error: ' + err } },
```

Comment:
tip: You can use `await` if you wrap in `async` function as is done in https://github.com/grafana/k6/blob/4615f0a1ed1b45fa16b83f7157349a4943dc1a05/js/modules/k6/http/async_request_test.go#L12-L15

Or did you have a different reason to not using `async`/`await`

Reply:
Good catch. I used the promise syntax out of convenience and because it helped avoid `try/catch` block. But as I've found out, you can use `.catch` on the result of an `await` call, which helps keep the code nice and tidy. Thus, I've ported the `wrapInAsyncLambda` helper, and used `async/await` instead 🤝